knowledge/terminology.md

```diff
@@ -2,15 +2,17 @@
-- **Black Magic Probe** (abbrev. **BMP**) - The “**native**” Black Magic Debug (BMD) hardware that was specifically designed and intended for the BMD project. This is the hardware you can buy from 1BitSquared, Adafruit and other resellers as “the Black Magic Probe”.
-- **Black Magic Core** (Abbrev. BMC) - This describes the core code of the project that handles JTAG/SWD protocol, implements Flashing, target detection and all target specific functionality. It does not include host platform specific code like for example code that toggles GPIO.
+- **Black Magic Firmware** (abbrev. **BMF**) or **Black Magic Debug Firmware** (abbrev. **BMDF**) - More specifically just the Black Magic Firmware that is loaded either onto Black Magic Probe hardware or onto any other supported host hardware.
+- **Black Magic App** (abbrev. **BMA**) or **Black Magic Debug App** (abbrev. **BMDA**) (formerly known as **Hosted** or **PC-Hosted**) - The Black Magic Debug system compiled as a PC application instead of a Firmware. Allowing it to be used in software in the middle mode instead of being contained to a hardware device. This used to be called “**hosted**” in the context of the project.
+- **Black Magic Probe** (abbrev. **BMP**) or **Black Magic Debug Probe** (abbrev. **BMDP**) - The “**native**” Black Magic Debug (BMD) hardware that was specifically designed and intended for the BMD project. This is the hardware you can buy from 1BitSquared, Adafruit and other resellers as “the Black Magic Probe”.
+- **Black Magic Core** (abbrev. **BMC**) or **Black Magic Debug Core** (abbrev. **BMDC**) - This describes the core code of the project that handles JTAG/SWD protocol, implements Flashing, target detection and all target specific functionality. It does not include host platform specific code like for example code that toggles GPIO.
 - **Host Platform** - Refers to the hardware that the BMD software (either Firmware or Application) runs on. It can be the PC, STM32, lm4f or some more specific hardware like Black Magic Probe aka. native, ST-Link, sw-link and so on.
-- **Probe** aka. **Cable** - Refers to the hardware that is connected between the PC running GDB (and/or BMA) and the Target (DUT). In the simplest case it is a USB to JTAG/SWD converter like an FTDI chip but in most cases it is a dedicated hardware running either our BMF or some other firmware.
+- **Probe** a.k.a. **Cable** - Refers to the hardware that is connected between the PC running GDB (and/or BMA) and the Target. In the simplest case it is a USB to JTAG/SWD converter like an FTDI chip but in most cases it is a dedicated hardware running either our BMF or some other firmware.
```

Comment:
"aka" is correct, having the dots between the letters is considered anachronistic in modern English.

Reply:
Resolved. Thanks! I was triggered by the dot at the end of `aka.`